File: go-sdk/pkg/models/operations/createadeployrequest.go
package operations

type CreateADeployRequestPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type CreateADeployRequestRequestBody struct {
	Branch     *string `json:"branch,omitempty"`
	IntoBranch *string `json:"into_branch,omitempty"`
	Notes      *string `json:"notes,omitempty"`
	Number     *string `json:"number,omitempty"`
}

type CreateADeployRequestRequest struct {
	PathParams CreateADeployRequestPathParams
	Request    *CreateADeployRequestRequestBody `request:"mediaType=application/json"`
}

type CreateADeployRequestResponse struct {
	ContentType                                  string
	StatusCode                                   int64
	CreateADeployRequest200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/createadeployrequest.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Type } from "class-transformer";


export class CreateADeployRequestPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class CreateADeployRequestRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "branch" })
  branch?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "into_branch" })
  intoBranch?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "notes" })
  notes?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "number" })
  number?: string;
}

export class CreateADeployRequestRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: CreateADeployRequestPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: CreateADeployRequestRequestBody;
}

export class CreateADeployRequestResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  createADeployRequest200ApplicationJSONObject?: Record<string, Record<string, any>>;
}